src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -848,50 +864,64 @@ public Type DfsTraverse(Node n)
         private LocalBuilder DeclareLocal(Type t, string identifier)
         {         
             if (compilePass == CompilePass.GatherTypeInfo) return null;
+#if DEBUG
             writer.WriteLine($"{nameof(ilGen.DeclareLocal)} {t} {identifier}");
+#endif
             return ilGen.DeclareLocal(t);
         }
 
         private void EmitOpCode(OpCode opCode, Label label)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode, label);
+#if DEBUG
             writer.WriteLine($"{opCode} {label}");
+#endif
         }
 
         private void EmitOpCode(OpCode opCode, LocalBuilder local)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode, local);
+#if DEBUG
             writer.WriteLine($"{opCode} {local}");
+#endif
         }
 
         private void EmitOpCode(OpCode opCode)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode);
+#if DEBUG
             writer.WriteLine(opCode);
+#endif
         }
 
         private void EmitOpCode(OpCode opCode, Type t)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode, t);
+#if DEBUG
             writer.WriteLine($"{opCode} {t}");
+#endif
         }
 
         private void EmitOpCode(OpCode opCode, int index)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode, index);
+#if DEBUG
             writer.WriteLine($"{opCode} {index}");
+#endif
         }
 
         private void EmitOpCode(OpCode opCode, string str)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode, str);
+#if DEBUG
```

Comment:
I guess ideally we would wrap this 3 lines into a function? maybe a tiny local one?